tests/src/test/scala/ch/renku/acceptancetests/pages/LoginPage.scala

```diff
@@ -97,69 +97,96 @@ case object LoginPage extends RenkuPage {
+    }
+
+    def confirmPasswordField(implicit webDriver: WebDriver): WebElement = eventually {
+      find(cssSelector("input#password-confirm")) getOrElse fail("Confirm Password field not found")
+    }
+
+    def registerButton(implicit webDriver: WebDriver): WebElement = eventually {
+      find(cssSelector("#kc-form-buttons > input")) getOrElse fail("Register button not found")
+    }
+  }
+
+  case class AuthorizeApplicationPage()
+      extends selenium.Page
+      with ScalatestMatchers
+      with Eventually
+      with AcceptanceSpecPatience {
+
+    final override lazy val url: String = "https://dev.renku.ch/gitlab/oauth/authorize"
+
+    def authorize(implicit webDriver: WebDriver): Unit =
+      authorizeButton.click()
+
+    def authorizeButton(implicit webDriver: WebDriver): WebElement = eventually {
+      find(
+        cssSelector(
+          "#content-body > main > div > div > div.modal-body > div > form:nth-child(2) > input.btn.btn-success.prepend-left-10"
+        )
+      ) getOrElse fail("Authorize button not found")
     }
   }
 
-  // An example of how to take screenshots during a run to get better debugging information
-  // def debugLogInWith(userCredentials: UserCredentials, spec: AcceptanceSpec)(implicit webDriver: WebDriver): LoginType =
-  //   eventually {
-  //     if (userCredentials.provider.value != "") {
-  //       println("use provider")
-  //       oidcButton.click() sleep (5 seconds)
-  //       val providerLoginPage = ProviderLoginPage(userCredentials)
-  //       if (!(currentUrl startsWith providerLoginPage.url)) {
-  //         println("Not on provider")
-  //         spec.saveScreenshot()
-  //         oidcButton.click() sleep (5 seconds)
-  //         spec.saveScreenshot()
-  //       }
-  //       providerLoginPage.logIn sleep (5 seconds)
-  //       LoginWithProvider
-  //     } else {
-  //       usernameField.clear() sleep (1 second)
-  //       usernameField.sendKeys(userCredentials.email.value) sleep (1 second)
-
-  //       passwordField.clear() sleep (1 second)
-  //       passwordField.sendKeys(userCredentials.password.value) sleep (1 second)
-
-  //       logInButton.click()
-
-  //       val providerLoginPage = ProviderLoginPage(userCredentials)
-  //       if (currentUrl startsWith providerLoginPage.url) {
-  //         providerLoginPage.logIn
-  //         LoginWithProvider
-  //       } else LoginWithoutProvider
-  //     }
-  //   }
+  def logInWith(userCredentials: UserCredentials)(implicit webDriver: WebDriver): Unit = eventually {
+    usernameField.clear() sleep (1 second)
+    usernameField.sendKeys(userCredentials.email.value) sleep (1 second)
+
+    passwordField.clear() sleep (1 second)
+    passwordField.sendKeys(userCredentials.password.value) sleep (1 second)
+
+    try {
+      logInButton.click()
+    } catch {
+      // Just try again
+      case e: StaleElementReferenceException => logInButton.click()
```

Comment:
Maybe the `sleep` here, too?

Reply:
I think this catch is not necessary at all. I've removed it to test that hypothesis. If it is necessary, I will add it back and include a sleep.

Reply:
👍